ql/src/java/org/apache/hadoop/hive/ql/ddl/table/AbstractBaseAlterTableAnalyzer.java

```diff
@@ -95,11 +98,16 @@ protected void addInputsOutputsAlterTable(TableName tableName, Map<String, Strin
       // ReadEntity as no lock.
       re.noLockNeeded();
       inputs.add(re);
+      Partition partition;
 
       if (AlterTableUtils.isFullPartitionSpec(table, partitionSpec)) {
         // Fully specified partition spec
-        Partition part = PartitionUtils.getPartition(db, table, partitionSpec, true);
-        outputs.add(new WriteEntity(part, writeType));
+        if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
```

Comment:
why not to make `PartitionUtils.getPartition(db, table, partitionSpec, true)` generic. it should return partitions based on table type

Reply:
Done